### ISSUE TITLE
BatchNormalization(mode=2) on keras 2.1.5

### ISSUE THREAD

**Issue by kristosh** [r0nn13/conditional-dcgan-keras#7]
In the sampled code for the keras implementation in the generator there is the batchnormalization step that it is coded as follow:

`e1 = BatchNormalization(mode=2)(e1)`

However, when the mode = 2 then there is an emerging error from keras. I noticed that when i set the mode=0 the whole thing is working. However, what it the difference between mode=2 and mode =0. And if it is crucial difference how can I have mode = 2. I guess that error has to do with the keras version (I am using keras 2.1.5).
